### PR TITLE
[BACKLOG-30195] Fixing memory leaks.

### PR DESCRIPTION
**Pull request by dcleao**
* Tab poling would leak global frame variables and yet bring no observable improvements.
* Fix memory leaks in `IFrameWatcher` code.

@pentaho/millenniumfalcon please review.

Merge after:
* https://github.com/pentaho/pentaho-commons-gwt-modules/pull/731
* https://github.com/pentaho/pentaho-analyzer/pull/1984

### CODE REVIEW

File: user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java [hunk 1]
```diff
@@ -190,83 +190,118 @@ protected void onAttach() {
...
-          wind.onunload = unloader;
-          wind.mantleEventsIn = true;
-          $wnd.watchWindow = null;
-        } catch(e){
-          //You're most likely here because of Cross-site scripting permissions... consuming
-        }
+      // IFrame's window instance.
+      var iframeWindow = frameElem.contentWindow;
+
+      // console.log("IFrameTabPanel.CustomFrame.attachEventListeners url=" + iframeWindow.location.href + " name=" + iframeWindow.name);
```
Comment (davidmsantos90) [inline user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java]: was this left intentionally or debug leftovers?

File: user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java [hunk 2]
```diff
@@ -190,83 +190,118 @@ protected void onAttach() {
...
+
+      function onChildWindowMouseEvent(event) {
+        // console.log("IFrameTabPanel.CustomFrame.onChildWindowMouseEvent url=" + this.location.href + " name=" +  this.name);
```
Comment (davidmsantos90) [inline user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java]: same

File: user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java [hunk 3]
```diff
@@ -190,83 +190,118 @@ protected void onAttach() {
...
+        //CHECKSTYLE IGNORE LineLength FOR NEXT 1 LINES
+        frame.@org.pentaho.mantle.client.solutionbrowser.tabs.IFrameTabPanel$CustomFrame::removeEventListeners(Lcom/google/gwt/dom/client/Element;)(frameElem);
+
+        startIFrameWatcher(iframeWindow);
       }
       
       // Starts the watching loop.
-      $wnd.startIFrameWatcher = function(wind){
+      function startIFrameWatcher(wind) {
+        // console.log("IFrameTabPanel.CustomFrame.startIFrameWatcher url=" + wind.location.href + " name=" +  wind.name);
```
Comment (davidmsantos90) [inline user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java]: same here

File: user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java [hunk 4]
```diff
@@ -190,83 +190,118 @@ protected void onAttach() {
...
-        $wnd.setTimeout("rehookEventsTimer()", 300);
+        $wnd.setTimeout(rehookEventsTimer, 300);
       }
     
-      // loop that's started when an iFrame unloads, when the url changes it adds back in the hooks
-      $wnd.rehookEventsTimer = function(){
+      // Loop that's started when an IFrame unloads.
+      // When the url changes it adds back in the hooks.
+      function rehookEventsTimer() {
+        // console.log("IFrameTabPanel.CustomFrame.rehookEventsTimer url=" + ($wnd.watchWindow && $wnd.watchWindow.location.href) + " name=" + ($wnd.watchWindow && $wnd.watchWindow.name));
```
Comment (davidmsantos90) [inline user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java]: same here

File: user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java [hunk 5]
```diff
@@ -190,83 +190,118 @@ protected void onAttach() {
...
-      function rehookEventsTimer(){
-        $wnd.rehookEventsTimer();
+
+      // Hooks up mouse and unload events.
+      function hookEvents(wind) {
+        if(wind == null) {
+            wind = $wnd.watchWindow;
+        }
+
+        // console.log("IFrameTabPanel.CustomFrame.hookEvents url=" + (wind && wind.location.href) + " name=" + (wind && wind.name));
```
Comment (davidmsantos90) [inline user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java]: same here

File: user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java (hunk 1 above)
Comment (dcleao) [reply user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java]: I left this on purpose, to add in any future debugging sessions. Debugging GWT code is particularly hard. Comments are removed if the PRETTY output format is not used, which is the case for production builds.

**Pull request merged**